plat/renesas/rcar/rcar_dvfs.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2016,2017 ARM Limited and Contributors. All rights reserved.
+ * Copyright (c) 2015-2017 Renesas Electronics Corporation. All rights reserved.
+ * Copyright (c) 2017 EPAM Systems Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * Neither the name of ARM nor the names of its contributors may be used
+ * to endorse or promote products derived from this software without specific
+ * prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "rcar_def.h"
+#include "rcar_private.h"
+
+struct op_points
+{
+	uint32_t freq;
+	uint32_t voltage;
+} rcar_op_points[] = {
+	{500,  820}, /* Hz, mV */
```

Comment:
500 Hz? Are you sure?

BTW, if you holding frequency in Hz, you can store frequencies up to 4GHz. Are you sure that this will be enough in next 10 years?

Reply:
Wrong comment. Should be /* MHz, mV */

Reply:
Okay, but what about 4GHz limit?

Reply:
The protocol requirement for frequencies to be in Hertz (Hz). We just follow it. I think when we come to this limitation (> 4GHz) there will be either yet another protocol or a modified version of this protocol to handle such cases